File: src/main/java/pl/training/workshop/m7/s10_booleanparameter/step3/MobileApp.java
package pl.training.workshop.m7.s10_booleanparameter.step3;

import java.util.Objects;

/**
 * Krok 3: klient zmigrowany - wywołanie mówi "online, okulary z kina"
 * bez zaglądania do sygnatury.
 */
public final class MobileApp {
    private final TicketService tickets;

    public MobileApp(TicketService tickets) {
        this.tickets = Objects.requireNonNull(tickets, "tickets");
    }

    public String buy(String title, String format, int seats) {
        return tickets.bookOnline(title, format, seats, Glasses.RENTED);
    }
}

File: csharp/src/Training.Workshop/M7/S10BooleanParameter/Step3/MobileApp.cs
namespace Training.Workshop.M7.S10BooleanParameter.Step3;

/// <summary>Krok 3: klient zmigrowany - wywołanie mówi "online, okulary z kina" bez zaglądania do sygnatury.</summary>
public sealed class MobileApp
{
    private readonly TicketService _tickets;

    public MobileApp(TicketService tickets)
    {
        ArgumentNullException.ThrowIfNull(tickets);
        _tickets = tickets;
    }

    public string Buy(string title, string format, int seats)
    {
        return _tickets.BookOnline(title, format, seats, Glasses.Rented);
    }
}
